feature-libs/asm/components/customer-list/customer-list.component.ts

```diff
@@ -0,0 +1,243 @@
+import { Component, OnInit } from '@angular/core';
+import { AsmConfig, AsmService } from '@spartacus/asm/core';
+import {
+  CustomerListColumnActionType,
+  CustomerListsPage,
+  CustomerSearchOptions,
+  CustomerSearchPage,
+} from '@spartacus/asm/root';
+import { SortModel, TranslationService, User } from '@spartacus/core';
+import {
+  BREAKPOINT,
+  BreakpointService,
+  FocusConfig,
+  ICON_TYPE,
+  ModalService,
+} from '@spartacus/storefront';
+import { combineLatest, Observable, of } from 'rxjs';
+import { filter, map, switchMap, tap } from 'rxjs/operators';
+import { CustomerListActionEvent } from './customer-list.model';
+
+@Component({
+  selector: 'cx-customer-list',
+  templateUrl: './customer-list.component.html',
+})
+export class CustomerListComponent implements OnInit {
+  private PAGE_SIZE = 5;
+
+  focusConfig: FocusConfig = {
+    trap: true,
+    block: true,
+    autofocus: 'customer-list-selector',
+    focusOnEscape: true,
+  };
+
+  iconTypes = ICON_TYPE;
+
+  BREAKPOINT = BREAKPOINT;
+
+  selectedUserGroupId: string | undefined;
+
+  customerSearchPage$: Observable<CustomerSearchPage | null>;
+
+  customerListsPage$: Observable<CustomerListsPage | undefined>;
+
+  selectedCustomer: User;
+
+  currentPage = 0;
+
+  maxPage = 0;
+
+  loaded = false;
+
+  sorts: SortModel[] | null;
+
+  sortCode: string | undefined;
+
+  breakpoint$: Observable<BREAKPOINT>;
+
+  customerListConfig = this.asmConfig?.asm?.customerList;
+
+  constructor(
+    protected modalService: ModalService,
+    protected asmService: AsmService,
+    protected breakpointService: BreakpointService,
+    protected asmConfig: AsmConfig,
+    protected translation: TranslationService
+  ) {
+    this.breakpoint$ = this.getBreakpoint();
+  }
+
+  ngOnInit(): void {
+    this.customerListsPage$ = this.asmService
```

Comment:
we can move this logic out of `ngOnInit`. So, just `customerListsPage$: Observable<CustomerListsPage | undefined> = this.asmService.getCustomerLists()....`

Reply:
I would prefer not to implement this change. This can make the component more involved to unit test.  

Assume the component initialization is done during class construction, per the suggestion.  In the unit test, the component class is typically already constructed by the time a specific unit test or describe block becomes active.  If you need to change the return values of spies for that test/describe, you need to re-construct the component so that the new values are used in the component initialization.